### PR TITLE
User proto utility functions - converting to/from strings

### PR DESCRIPTION
**Pull request by willvedd**
## Description
This PR introduces two utility functions:
 - Converting a `openfgav1.User` protobuf type to a string
 - Converting a string to a `openfgav1.User` protobuf type
 
 These utility functions aid in unit testing and eventually for creating cache keys to deduplicate results from `ListUsers`. 

This was suggested in a [PR comment](https://github.com/openfga/openfga/pull/1448#discussion_r1526621107) and was initially implemented in #1457 but cherry-picked for more focused review.
 

## References
- Cherry-picked from #1457
- Related PR comment: https://github.com/openfga/openfga/pull/1448#discussion_r1526621107
- [ListUsers Protobuf PR](https://github.com/openfga/api/pull/144)

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/tuple/tuple.go [hunk 1]
```diff
@@ -154,6 +154,51 @@ func ObjectKey(obj *openfgav1.Object) string {
 	return fmt.Sprintf("%s:%s", obj.GetType(), obj.GetId())
 }
 
+type UserString = string
+
+// UserProtoToString returns a string from a User proto. Ex: 'user:maria#member'. It is
+// the opposite from StringToUserProto function.
+func UserProtoToString(obj *openfgav1.User) UserString {
```
Comment (miparnisari) [inline pkg/tuple/tuple.go:161]:
```suggestion
// UserProtoToString returns a string from a User proto. Ex: 'user:maria' or 'group:fga#member'. It is
// the opposite from StringToUserProto function.
func UserProtoToString(obj *openfgav1.User) UserString {
```

File: pkg/tuple/tuple_test.go [hunk 2]
```diff
@@ -360,3 +360,107 @@ func TestGetObjectRelationAsString(t *testing.T) {
...
 }
+
+func TestFromUserProto(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		input    *openfgav1.User
+		expected string
+	}{
+		{
+			name: "user_with_relation",
```
Comment (miparnisari) [inline pkg/tuple/tuple_test.go:371]: there is no relation in this test case

File: pkg/tuple/tuple_test.go [hunk 3]
```diff
@@ -360,3 +360,107 @@ func TestGetObjectRelationAsString(t *testing.T) {
...
+}
+
+func TestToUserProto(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		input    string
+		expected *openfgav1.User
+	}{
+		{
+			name:  "user_with_relation",
```
Comment (miparnisari) [inline pkg/tuple/tuple_test.go:423]: same here

Comment (willvedd) [reply pkg/tuple/tuple_test.go]: 🤦‍♂️ typo

**Review by miparnisari — APPROVED** [review]

**Pull request merged**